### ISSUE TITLE
Inherit shower-core ESLint config

### ISSUE THREAD

**Issue by pepelsbey** [shower/cli#18]
It would be good to have common JS code style throughout the whole organization.

If you don’t mind, I’d suggest using `;` and 4 spaces, just like we do in [.eslintrc.yml](https://github.com/shower/core/blob/master/.eslintrc.yml) of shower-core, the main code base of the Shower project. It’s basically `airbnb-base` with some extras. The same, but in a form if [.editorconfig](.editorconfig) and [.stylelintrc](https://github.com/shower/material/blob/master/.stylelintrc#L104) exists in both themes.

We can always discuss details, but having a common code style is essential for a project like this.

**Comment by pepelsbey** [comment]
You could do that? :O

**Comment by shvaikalesh** [comment]
[Latest version here.](https://github.com/shower/core/blob/merge-player-with-container/.eslintrc.yml)
I am not supper happy with how airbnb config is unnecessary restrictive, perhaps there is a popular config we can use with less overrides. Definitely 👍 on extracting the config though. I think we should start using scoped packages: `@shower/eslint-config`, `@shower/core`, or consider making it `pepelsbey-eslint-config`.

We can also evaluate prettier: it allows 4 spaces and bracket spacing.

**Comment by pepelsbey** [comment]
Yes, I’m all for `@shower` scoping, but it takes some time to update all packages and I just haven’t done anything like this before :(

**Comment by nikolay-govorov** [comment]
If we want to create `@shower` scope, maybe will be conveniently using one eslint config in main repository?

**Comment by nikolay-govorov** [comment]
I can create `shower-config`, but do we need this if we go to scope?

**Comment by shvaikalesh** [comment]
Resolution of offline conversation with @pepelsbey: let's hold off with extracting the config and give `prettier` a try (right after we ship `@shower/core@3`). We will still need ESLint, but with far less rules and, hopefully, overrides.

**Comment by avdeev** [comment]
I seems to me that the `eslint --fix` works like a `prettier`.

**Comment by nikolay-govorov** [comment]
Not sure, but eslint is much more powerful. Prettier is designed only for formatting, but it does not replace a full-fledged Linter.

**Comment by nikolay-govorov** [comment]
Hey. I tried the eslint+prettier bundle that the kernel uses. Sometimes she does strange things:

Such code:
```js
const semver = require('semver');

const pkg = require('../package.json');

if (!semver.satisfies(process.version, pkg.engines.node)) {
    /* ... */
}
```

will turn into such:
```js
const semver = require('semver');

if (!semver.satisfies(process.version, pkg.engines.node)) {
    /* ... */
}

const pkg = require('../package.json');
```

In automatic mode, there is no guarantee that the code will not be damaged, and the only benefit from prettier — automatic formatting. Now I propose to return to the original version with a common eslint config based on the [@shower/core](https://github.com/shower/core) config.

**Comment by nikolay-govorov** [comment]
In the opposite direction, I really want to keep eslint — it is a static analysis tool, it protects against many potential errors. Prettier won't be able to replace it.

**Comment by shvaikalesh** [comment]
@nikolay-govorov `require` are getting sorted by [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md) that came with [Airbnb config](https://www.npmjs.com/package/eslint-config-airbnb), not Prettier.

**Comment by nikolay-govorov** [comment]
@shvaikalesh, Yes, you're right — I don't understand why this is happening yet, but it's not Prettier's fault.
